Pinta.Gui.Widgets/Widgets/Canvas/PintaCanvas.cs

```diff
@@ -332,6 +333,74 @@ private bool ShouldShowCanvasGrid ()
 		return minViewDistance >= MIN_GRID_LINE_DISTANCE;
 	}
 
+	private void DrawCanvasAxonometricGrid (Gtk.Snapshot snapshot, Graphene.Rect canvasViewBounds)
+	{
+		if (!ShouldShowCanvasAxonometricGrid ())
+			return;
+
+		Gsk.Path gridPath = BuildCanvasAxonometricGridPath ();
+
+		snapshot.PushClip (canvasViewBounds);
+		snapshot.Save ();
+
+		// Scale the selection path up to the view size.
+		float scale = (float) document.Workspace.Scale;
+		snapshot.Scale (scale, scale);
+
+		// Draw as a dotted line (every other pixel) to have a more subtle appearance.
+		Gsk.Stroke stroke = Gsk.Stroke.New (lineWidth: 1.0f / scale);
+		stroke.SetDash ([1.0f / scale, 1.0f / scale]);
+
+		Gdk.RGBA color = new () { Red = 0, Green = 0, Blue = 0, Alpha = 1 };
+		snapshot.AppendStroke (gridPath, stroke, color);
+
+		snapshot.Restore ();
+		snapshot.Pop ();
+	}
+
+	private Gsk.Path BuildCanvasAxonometricGridPath ()
+	{
+		int axonometricWidth = canvas_grid.AxonometricWidth;
+		int imageHeight = document.ImageSize.Height;
+		int imageWidth = document.ImageSize.Width;
+
+		Gsk.PathBuilder pathBuilder = Gsk.PathBuilder.New ();
+		// Add diagonal lines.
+		for (int i = 0; i < imageWidth + imageHeight; i += 2 * axonometricWidth) {
+			pathBuilder.MoveTo (0, i);
+			pathBuilder.LineTo (i, 0);
+		}
+
+		for (int i = -imageHeight + imageHeight % (2 * axonometricWidth); i < imageWidth; i += 2 * axonometricWidth) {
```

Comment:
Maybe some of these expressions could be factored out into variables? This loop statement is fairly complex to read

Reply:
Added some comments to make intent more clear. Variables are not needed here because the code is quite straightforward and variables would only pollute more, and should be even easier to understand with the new comments.